embassy-stm32/src/lib.rs

```diff
@@ -296,3 +331,41 @@ pub fn init(config: Config) -> Peripherals {
         p
     })
 }
+
+#[cfg(ucpd)]
+/// Safety: must only be called when all UCPDs are disabled (e.g. at startup)
+unsafe fn ucpd_init(
+    _cs: critical_section::CriticalSection,
+    ucpd1_db_enable: bool,
+    #[cfg(stm32g0x1)] ucpd2_db_enable: bool,
+) {
+    #[cfg(all(ucpd, stm32g0x1))]
```

Comment:
no need to check for `ucpd` here if the function is `#[cfg(ucpd)]`

Reply:
good catch 😅